### PR TITLE
Experiment/tornado: Ldap Authentication

### PR DESCRIPTION
**Pull request by untra**
- Found a means to successfully authenticate against CU's ldap server. Currently it can just tell if the username / password combination is valid, but that's the hard part. Soon we should then be able to pull valid info from ldap records into our system.
- Priority should be attached to getting SSL setup. If we are authenticating, we shouldn't be doing it over plaintext
- shuffled the main function around

### CODE REVIEW

File: ldapauth.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+pip install python-ldap
+"""
+import ldap
+import logging
+
+# where to start the search for users
+LDAP_SEARCH_BASE = 'dc=colorado,dc=edu'
+
+# the server to auth against
+LDAP_URL = 'ldap://directory.colorado.edu'
```
Comment (antsankov) [inline ldapauth.py:16]: Is there we would use an HTTPS connection?

**Comment by antsankov** [comment]
I like it +1.

File: ldapauth.py (hunk 1 above)
Comment (untra) [inline ldapauth.py:16]: we need a cert to authenticate over tls

**Comment by untra** [comment]
I want to get this merged in soon. Lets look this over, someone, yeah?

**Comment by antsankov** [comment]
This is being merged on principle.

**Pull request merged**